test/jdk/java/nio/file/Files/SBC.java

```diff
@@ -424,6 +425,14 @@ static void nullTests(Path dir) throws Exception {
         } catch (NullPointerException x) { }
     }
 
+    static void emptyPathTest() throws Exception {
+        try {
+            OpenOption[] opts = { WRITE, CREATE_NEW };
+            Files.newByteChannel(Path.of(""), opts);
+            throw new RuntimeException("FileAlreadyExistsException expected");
+        } catch (FileAlreadyExistsException x) { }
```

Comment:
While WRITE + CREATE_NEW are the options to tickle the bug, I think emptyPathTest should be expanded to test over combinations of open options. There are several code paths with options such as O_NOFOLLOW and DELETE_ON_CLOSE that will also need to be tested.

Reply:
I extended the test a bit. Have a look, please.

Reply:
Indeed, on Windows the WinAPI function used to open a file (`CreateFileW`) will return `ERROR_ACCESS_DENIED` when used on a directory _without_ a special flag (`FILE_FLAG_BACKUP_SEMANTICS`). This flag is not specified in this particular code path. I will make the necessary changes to the test.